### PR TITLE
Rename CreateDirectory to CitusCreateDirectory

### PR DESCRIPTION
**Pull request by lithp**

### CODE REVIEW

File: src/backend/distributed/worker/worker_partition_protocol.c [hunk 1]
```diff
@@ -602,7 +602,7 @@ DirectoryExists(StringInfo directoryName)
 
 /* Creates a new directory with the given directory name. */
 void
-CreateDirectory(StringInfo directoryName)
+CitusCreateDirectory(StringInfo directoryName)
```
Comment (mtuncer) [inline src/backend/distributed/worker/worker_partition_protocol.c]: Are we going to put ```Citus``` prefix to our general purpose functions ? 

I suggest we have a common consensus on this one as we might do this multiple times as we go.

Comment (jasonmp85) [reply src/backend/distributed/worker/worker_partition_protocol.c]: FWIW I was always in favor of using visibility modifiers to whitelist symbols that we actually need to be visible.

**Comment by mtuncer** [comment]
@ozgune any opinion on this ?

File: src/backend/distributed/worker/worker_partition_protocol.c (hunk 1 above)
Comment (marcocitus) [reply src/backend/distributed/worker/worker_partition_protocol.c]: > Are we going to put Citus prefix to our general purpose functions ?

Seems reasonable when there is a name collision. We did this for a long time with, e.g. `CitusNodeToString`.

Comment (ozgune) [reply src/backend/distributed/worker/worker_partition_protocol.c]: I agree with @marcocitus on this. We'll only need to make this change for three functions.

**Pull request merged**